packages/components/nodes/chatmodels/ChatGoogleGenerativeAI/ChatGoogleGenerativeAI.ts

```diff
@@ -98,10 +169,67 @@ class GoogleGenerativeAI_ChatModels implements INode {
 
         const model = new ChatGoogleGenerativeAI(obj)
         if (topP) model.topP = parseFloat(topP)
+        if (topK) model.topP = parseFloat(topK)
         if (cache) model.cache = cache
         if (temperature) model.temperature = parseFloat(temperature)
+
+        // Safety Settings
+        let harmCategories: string[] = convertMultiOptionsToStringArray(harmCategory)
+        let harmBlockThresholds: string[] = convertMultiOptionsToStringArray(harmBlockThreshold)
+        if (harmCategories.length != harmBlockThresholds.length)
+            throw new Error(`Harm Category & Harm Block Threshold are not the same length`)
+        const safetySettings: SafetySetting[] = harmCategories.map((value, index) => {
+            return {
+                category: categoryInput(value),
+                threshold: thresholdInput(harmBlockThresholds[index])
+            }
+        })
+        if (safetySettings.length > 0) model.safetySettings = safetySettings
+
         return model
     }
 }
 
+const categoryInput = (categoryInput: string): HarmCategory => {
+    let categoryOutput: HarmCategory
+    switch (categoryInput) {
+        case HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT:
+            categoryOutput = HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT
+            break
+        case HarmCategory.HARM_CATEGORY_HATE_SPEECH:
+            categoryOutput = HarmCategory.HARM_CATEGORY_HATE_SPEECH
+            break
+        case HarmCategory.HARM_CATEGORY_HARASSMENT:
+            categoryOutput = HarmCategory.HARM_CATEGORY_HARASSMENT
+            break
+        case HarmCategory.HARM_CATEGORY_SEXUALLY_EXPLICIT:
+            categoryOutput = HarmCategory.HARM_CATEGORY_SEXUALLY_EXPLICIT
+            break
+        default:
+            categoryOutput = HarmCategory.HARM_CATEGORY_UNSPECIFIED
+    }
+    return categoryOutput
+}
```

Comment:
switch statement that is somewhat redundant because each case simply assigns the same value that was input to categoryOutput.

```typescript
const categoryInput = (input: string): HarmCategory => {
    return Object.values(HarmCategory).includes(input as any) ? input as HarmCategory : HarmCategory.HARM_CATEGORY_UNSPECIFIED;
}
```

Reply:
This is great. Great learn 👍

Didn't know I can straight use `as` to change the type to `HarmCategory `